### ISSUE TITLE
Text offset from marker when using VectorTile layer

### ISSUE THREAD

**Issue by dragonnn** [openlayers/openlayers#8213]
- [x] I am submitting a bug or feature request, not a usage question. Go to https://stackoverflow.com/questions/tagged/openlayers for questions.
- [x] I have searched GitHub to see if a similar bug or feature request has already been reported.
- [x] I have verified that the issue is present in the latest version of OpenLayers (see 'LATEST' on https://openlayers.org/). - I am using Version: v4.6.5
- [x] If reporting a bug, I have created a [CodePen](https://codepen.io) or prepared a stack trace (using the latest version and unminified code, so e.g. `ol-debug.js`, not `ol.js`) that shows the issue. https://codepen.io/dragonn/pen/NMJEGg?editors=0010
Zoom at 0,0, when the text appears I got effect like that:
![image](https://user-images.githubusercontent.com/1771469/40406388-c25578c6-5e60-11e8-93f7-3fdabaadbb36.png)
Now comment 
`map.addLayer(vectorLayer);`
And it will look like this:
![image](https://user-images.githubusercontent.com/1771469/40406415-e036462c-5e60-11e8-8d20-43fbc616605c.png)
The second without the vectorLayer is a desired effect. Sometimes the text is even cut:
![image](https://user-images.githubusercontent.com/1771469/40406679-ea91e742-5e61-11e8-8e18-b8c3a8e45659.png)
Screenshot from my application.

**Comment by ahocevar** [comment]
Can you strip that code pen down to the relevant code that shows the issue? With so much app code in the code pen, no one will be able to isolate the problem. Thanks.

**Comment by dragonnn** [comment]
Not really, I tried to replace my vectorLayer with https://openlayers.org/en/latest/examples/osm-vector-tiles.html but the problem doesn't exist then. 
My code is heavy based on this https://openlayers.org/en/latest/examples/geojson-vt.html
But when I remove the style function from VectorTile it seams to work ok. So even removing that is not possible. 
https://codepen.io/dragonn/pen/vjMpBg This is the best I can strip down without losing the problem.
I hope my code isn't faulty but I don't think so, since it works ok without the VectorTile

**Comment by dragonnn** [comment]
Ok I find out something:
`text: new ol.style.Text({
          font: ''+font_size+'px sans-serif',
          text: map.getView().getZoom() > 16 ?  feature.get('Text') || '': '',
          fill: new ol.style.Fill({color: '#111'}),
          stroke: new ol.style.Stroke({color: '#fff', width: 3*scale}),
        }),`
Removing that style from VectorTile fixes the problem. But how can a style on one layer affect another layer?

**Comment by ahocevar** [comment]
This is indeed an issue with your application property. The `font` property (`bold 11px`) for your vector layer style is invalid. According to https://developer.mozilla.org/en-US/docs/Web/CSS/font, the `font` property *must* contain `font-size` and `font-family`. So if you change the font propety to `bold 11px sans-serif`, things work as expected.

**Comment by dragonnn** [comment]
Oh, thank you. Confusing that the problem appeared when I started using VectorTile layer.
